src/script/event/preprocessor/EventStorageMiddleware/types.ts

```diff
@@ -27,7 +27,7 @@ import {IdentifiedUpdatePayload} from '../../EventService';
 export type HandledEvents = ClientConversationEvent | ConversationEvent;
 export type DBOperation =
   | {type: 'update'; event: HandledEvents; updates: IdentifiedUpdatePayload}
-  | {type: 'sequential-update'; event: EventRecord; updates: Partial<EventRecord>}
+  | {type: 'sequential-update'; event: HandledEvents; updates: IdentifiedUpdatePayload}
```

Comment:
I could have detected it just from the type level. It was weird to return a stored event at this point :)